### ISSUE TITLE
[BUG/ISSUE] Missing oxidant fields for Hg simulation via KPP (and other issues)

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#1192]
### What institution are you from?
GCST

### Description of the problem
I've noticed that some of the oxidant fields that are required for the new Hg simulation via KPP have not been added to the `HEMCO_Config.rc.Hg` template file.  Also, for some of these fields we do not have archived data yet.

The issues that I've found are:

1. In routine `Set_HcoPointers`, the code expects to read the `GLOBAL_fOA` oxidant field, but we do not have that on disk as part of our 10-year benchmark output:

```console
===============================================================================
GEOS-Chem ERROR: Cannot get pointer from HEMCO for Global fOA
 -> at Set_HCOPointers (in module GeosCore/mercury_mod.F90)
===============================================================================
```

2. `Set_HcoPointers` expected to read `GLOBAL_Br` and `GLOBAL_BrO` but the Br and BrO fields are listed in HEMCO_Config.rc under different container names:

```console
#===============================================================================
# --- BrOx fields --
#==============================================================================
(((BrOx_GC
* Br_GC          $ROOT/GCClassic_Output/13.0.0/$YYYY/GEOSChem.SpeciesConc.$YYYY$MM$DD_0000z.nc4      SpeciesConc_Br    2010-2019/1-12/1/0 C xyz 1        * - 1 1
* BrO_GC         $ROOT/GCClassic_Output/13.0.0/$YYYY/GEOSChem.SpeciesConc.$YYYY$MM$DD_0000z.nc4      SpeciesConc_BrO   2010-2019/1-12/1/0 C xyz 1        * - 1 1
)))BrOx_GC

(((.not.BrOx_GC
* Br_TOMCAT      $ROOT/MERCURY/v2014-09/BrOx/BrOx.GMI.geos5.2x25.nc                                  LBRO2N                 1985/1-12/1/0 C xyz pptv     * - 1 1
* BrO_TOMCAT     $ROOT/MERCURY/v2014-09/BrOx/BrOx.GMI.geos5.2x25.nc                                  LBRO2H                 1985/1-12/1/0 C xyz pptv     * - 1 1
))).not.BrOx_GC
```

3. Entries for other oxidant fields including `GLOBAL_CO`, `GLOBAL_CH4`, `GLOBAL_PM25`, and possibly others were missing from HEMCO_Config.rc.

These issues have been causing Hg integration tests to fail.

## Solution (partial)

I was able to solve issue 2 above by adding this code into `Set_HcoPointers`:
```F90
    ! Check if the Br_Ox switch is set
    thisOpt    = HCO_GetOpt( HcoState%Config%ExtList, 'BrOx_GC', ExtNr=0 )
    is_BrOx_GC = ( INDEX( thisOpt, 'true' ) > 0 )

    ! Do for each fixed KPP species
    DO N = 1, State_Chm%nKppFix

       ! Get species ID
       spcID    = State_Chm%Map_KppFix(N)

       ! Get oxidant name
       thisName = State_Chm%SpcData(SpcID)%Info%Name

       ! Construct field name using species name
       ! Br/BrO use a different container name than the other oxidant fields
       SELECT CASE( TRIM( thisName ) )
          CASE( 'Br' )
             IF ( is_BrOx_GC ) THEN
                fieldName = 'Br_GC'
             ELSE
                fieldName = 'Br_TOMCAT'
             ENDIF

          CASE( 'BrO' )
             IF ( is_BrOx_GC ) THEN
                fieldName = 'BrO_GC'
             ELSE
                fieldName = 'BrO_TOMCAT'
             ENDIF

          CASE DEFAULT
             fieldName = 'GLOBAL_' // TRIM( thisName )

       END SELECT
```
so that will pick the proper container names for Br and BrO.  This has been fixed in commit 6d59b850dbedd51c93c60d2a81e591c8d8d18cc7.

I was also able to add the missing entries for CO, CH4, and PM25 to the  HEMCO_Config.rc.Hg template file.  This was fixed in commit 818f25e6a49bb56682e1a719faff053a0502feb9.

## Action items

1. Need data files for GLOBAL_fOA and perhaps other missing oxidant fields
2. Would appreciate it if someone could beta-test the Hg simulation and supply fixes as needed.

Tagging @jennyfisher @cdholmes @msl3v @viral211 @yanxuz @msulprizio

**Comment by yantosca** [comment]
Also there is no entry in the `HEMCO_Config.rc.Hg` template file called `GLOBAL_RH`.
```F90
    !----------------------------------
    ! Relative humidity
    !----------------------------------
    ! Get pointer to this field.
    CALL HCO_GetPtr( HcoState, 'GLOBAL_RH', GLOB_RH, RC )
```

**Comment by yantosca** [comment]
Am wondering if we should just get RH from the State_Met object instead of reading it in from disk.

**Comment by viral211** [comment]
Hi @yantosca. The HEMCO_Config.rc file and the input data were also updated as part of the Hg chemistry updates. These updates have probably not been included in 13.4. You can find the necessary files here:  `/net/seasasfs02/srv/export/seasasfs02/share_root/vshah/Hg/rundirsNew` Please let me know when you have these in and I can test the simulation got any other missing items.

**Comment by yantosca** [comment]
Thanks @viral211.  I'll take a look at these as soon as I can.

**Comment by yantosca** [comment]
See PR #1196

**Comment by cdholmes** [comment]
Thanks for fixing these things. In case you haven't already done it, I support using RH from state_met.

**Comment by yantosca** [comment]
We can close this issue now that PR #1196 has been merged into the 13.4.0 development stream.
